bin/yb-ctl

```diff
@@ -640,6 +652,28 @@ class ClusterOptions:
         if self.enable_ysql is None:
             self.enable_ysql = True
 
+        # -----------------------------------------------------------------------------------------
+        # Client protocol ports
+        # -----------------------------------------------------------------------------------------
+
+        self.ysql_port = None
+        self.ycql_port = None
+        self.yedis_port = None
+        for api_name_str in ['ysql', 'ycql', 'yedis']:
```

Comment:
would it make sense to refactor the dict to be something like ports.api.type, where API could be master/tserver/yql/yedis/ysql and type could be http vs rpc?